src/Psl/Type/Exception/PathExpression.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Type\Exception;
+
+use Psl\Str;
+
+/**
+ * @psalm-immutable
```

Comment:
It could be both really..

The reason it got introduced is to make sure the path formats are the same across the repository. If people build their own custom types (which is a public API), they might as well end up in situations where they want to be using the path system. So I'dd say public API.

I've added documentation to the various methods.

Reply:
lets move this to `Internal\` and mark it internal, i do not see the need of having it as part of the public API.